### PR TITLE
Add event units tests

### PR DESCRIPTION
**Pull request by sebrandon1**
Similar to: #314

### CODE REVIEW

File: pkg/events/events.go [hunk 1]
```diff
@@ -9,24 +9,29 @@ import (
...
-	apiClient *clients.Settings
+	apiClient corev1Typed.EventInterface
 	// errorMsg used in discovery function before sending api request to cluster.
 	errorMsg string
 }
 
 // Pull pulls existing Event from cluster.
 func Pull(apiClient *clients.Settings, name, nsname string) (*Builder, error) {
+	if apiClient == nil {
+		return nil, fmt.Errorf("apiClient cannot be nil")
```
Comment (kononovn) [inline pkg/events/events.go:28]: Please add glog before return
glog.V(100).Infof("The apiClient is empty")

File: pkg/events/events.go [hunk 2]
```diff
@@ -36,22 +41,18 @@ func Pull(apiClient *clients.Settings, name, nsname string) (*Builder, error) {
 	}
 
 	if name == "" {
-		glog.V(100).Infof("The name of the Event is empty")
```
Comment (kononovn) [inline pkg/events/events.go:39]: Please don't remove glog

File: pkg/events/events.go [hunk 3]
```diff
@@ -36,22 +41,18 @@ func Pull(apiClient *clients.Settings, name, nsname string) (*Builder, error) {
...
-
-		builder.errorMsg = "Event 'name' cannot be empty"
+		return nil, fmt.Errorf("event 'name' cannot be empty")
 	}
 
 	if nsname == "" {
-		glog.V(100).Infof("The namespace of the Event is empty")
```
Comment (kononovn) [inline pkg/events/events.go:45]: Please don't remove glog

File: pkg/events/events_test.go [hunk 4]
```diff
@@ -1 +1,149 @@
...
+			}
+		} else {
+			assert.Nil(t, err)
+			assert.Equal(t, testEvent.Name, result.Object.Name)
+			assert.Equal(t, testEvent.Namespace, result.Object.Namespace)
+		}
+	}
+}
+
+func buildValidTestBuilder() *Builder {
```
Comment (kononovn) [inline pkg/events/events_test.go:93]: Please move private function to the end of the file

File: pkg/events/events_test.go [hunk 5]
```diff
@@ -1 +1,149 @@
...
+		apiClient: k8sfake.NewSimpleClientset().CoreV1().Events("test-namespace"),
+		Object: &corev1.Event{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-event",
+				Namespace: "test-namespace",
+			},
+		},
+	}
+}
+
```
Comment (kononovn) [inline pkg/events/events_test.go:104]: Can you add unit test for Exists function?

**Review by kononovn — CHANGES_REQUESTED** [review]
Please check few comments above.

File: pkg/events/events.go (hunk 1 above)
Comment (sebrandon1) [reply pkg/events/events.go:29]: Done.

File: pkg/events/events.go (hunk 2 above)
Comment (sebrandon1) [reply pkg/events/events.go:39]: Done.

File: pkg/events/events.go (hunk 3 above)
Comment (sebrandon1) [reply pkg/events/events.go:45]: Done.

File: pkg/events/events_test.go (hunk 5 above)
Comment (sebrandon1) [reply pkg/events/events_test.go:104]: Covered by Pull test.

File: pkg/events/events_test.go (hunk 4 above)
Comment (sebrandon1) [reply pkg/events/events_test.go:93]: Done.

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**